### PR TITLE
LibCpp: Do macro substitution in the preprocessor instead of the parser

### PR DESCRIPTION
**Pull request by itamar8910**

### CODE REVIEW

File: Userland/Libraries/LibCpp/Preprocessor.h [hunk 1]
```diff
@@ -38,13 +38,13 @@ class Preprocessor {
...
-    void handle_preprocessor_keyword(const StringView& keyword, GenericLexer& line_lexer);
+    PreprocessorKeyword handle_preprocessor_line(StringView const&);
+    void handle_preprocessor_keyword(StringView const& keyword, GenericLexer& line_lexer);
+    Vector<Token> process_line(StringView const& line);
 
-    Definitions m_definitions;
     const String m_filename;
-    const StringView m_program;
-    StringBuilder m_builder;
+    const String m_program;
```
Comment (alimpfard) [inline Userland/Libraries/LibCpp/Preprocessor.h:56]: `const String` seems odd, did you mean `const String&`?
(also above for the filename)

Comment (alimpfard) [reply Userland/Libraries/LibCpp/Preprocessor.h:56]: if you _did_, please consider switching to `StringView`

**Review by alimpfard — COMMENTED** [review]
Just a few random things I noticed

File: Userland/Libraries/LibCpp/Preprocessor.h (hunk 1 above)
Comment (itamar8910) [reply Userland/Libraries/LibCpp/Preprocessor.h]: Yeah the `const` on those Strings doesn't make much sense, I remove them.

**Pull request merged**